From 15755428aef2498a0f986af33f3243525daacaf9 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Thu, 29 Dec 2022 07:45:47 +0200
Subject: [PATCH] [SW-113135] mm: add sram partition for bootinfo

Starting gaudi3, bootinfo is used in ppboot which make use of ROM APIs.
However the ROM APIs for bootinfo is no-merge type hence ppboot retrieve
complete data from bootinfo to SRAM GP section, and then perform
set and finalize to add new boot info type. Currently it uses SRAM GP
section to store the bootinfo data. This patch adds new memory partition
for bootinfo which is kept as 256KB for gaudi3 and carved out of SRAM
GP section.

Change-Id: I0f8c1e433d70ecf5686e84ca61f8c5450f628b69
---

diff --git a/include/memory.h b/include/memory.h
index 5065555..3623d7c 100644
--- a/include/memory.h
+++ b/include/memory.h
@@ -12,7 +12,9 @@
 #include <common.h>
 
 #define MEM_GP_BASE		(SDRAM_BASE + 0x40000)	/* SRAM_BASE + 256K */
-#define MEM_GP_SZ		0xC0000			/* 768K */
+#define MEM_GP_SZ		0x80000			/* 512K */
+#define MEM_BOOT_INFO_BASE	(SDRAM_BASE + 0xC0000)	/* SRAM_BASE + 768K */
+#define MEM_BOOT_INFO_SZ	0x40000			/* 256K */
 #define MEM_LKD_PCI_BASE	(SDRAM_BASE + 0x200000)	/* SRAM_BASE + 2M */
 #define MEM_LKD_PCI_SZ		0x200000		/* 2M */
 #define MEM_FWUPD_PCI_BASE	(SDRAM_BASE + 0x600000)	/* SRAM_BASE + 6M */
@@ -38,6 +40,7 @@
 	MEM_FWUPD_PCI,
 	MEM_I2C,
 	MEM_SRAM_BIST,
+	MEM_BOOT_INFO,
 	MEM_NOENT	/* terminator */
 };
 
diff --git a/lib/preboot/memory.c b/lib/preboot/memory.c
index c917dca..b60acb4 100644
--- a/lib/preboot/memory.c
+++ b/lib/preboot/memory.c
@@ -33,6 +33,10 @@
 		.base = MEM_SRAM_BIST_DATA_BASE,
 		.size = MEM_SRAM_BIST_DATA_SZ,
 	},
+	[MEM_BOOT_INFO] = {
+		.base = MEM_BOOT_INFO_BASE,
+		.size = MEM_BOOT_INFO_SZ,
+	},
 };
 
 __weak int preboot_get_dev_spec_mem_region(enum mem_region region,
@@ -51,6 +55,7 @@
 	case MEM_FWUPD_PCI:
 	case MEM_I2C:
 	case MEM_SRAM_BIST:
+	case MEM_BOOT_INFO:
 		break;
 	default:
 		return 1;
